# Python Data Analysis Library

Le module glob recherche tous les chemins correspondant à un motif particulier selon les règles utilisées par le shell Unix, les résultats sont renvoyés dans un ordre arbitraire.
pandas est une bibliothèque open source sous licence BSD fournissant des structures de données hautes performances et faciles à utiliser, ainsi que des outils d'analyse des données pour le langage de programmation Python.

In [1]:
import glob
import pandas as pd

In [3]:
liste = glob.glob("C:/Users/Papa-SEYE/Documents/pandas tuto/*.csv")

In [9]:
liste

['C:/Users/Papa-SEYE/Documents/pandas tuto\\cmq.csv',
 'C:/Users/Papa-SEYE/Documents/pandas tuto\\militaires.csv']

Chaque élément de liste est un fichier csv donc on peut utiliser pandas pour lire ces données en python

df1 est le premier dataframe c'est à dire l'élément qui est à la position zéro dans liste. Et df2 est le deuxième dataframe

In [5]:
df1 = pd.read_csv(liste[0])

In [6]:
df2 = pd.read_csv(liste[1])

On peut utiliser la fonction head() pour les cinq premières lignes de notre dataframe. Un data frame est un tableau en deux dimensions: il contient des observations et des colonnes. C'est comme une table dans une base de données.

In [7]:
df1.head()

num  annee   prenom      nom genre          specialite1 specialite2  \
0  #30001   1930   Lucien    Amyot     M  Médecine de famille         NaN   
1  #30002   1930  Wilfrid    Audet     M             Inconnue         NaN   
2  #30003   1930    A. T.  Batshaw     M  Médecine de famille         NaN   
3  #30005   1930  Maurice  Beaudry     M         Dermatologie         NaN   
4  #30006   1930    P. E.  Belleau     M     Médecine interne         NaN   

  specialite3              statut dateChangStatut typePermis     ville  \
0         NaN  Retrait pour décès      1981-12-29   Régulier  Montréal   
1         NaN           Démission      1987-10-01   Régulier      Alma   
2         NaN  Retrait pour décès      1983-11-11   Régulier  Montréal   
3         NaN  Retrait pour décès      1989-02-12   Régulier    Québec   
4         NaN  Retrait pour décès      1986-05-31   Régulier    Québec   

     prov    pays  
0  Québec  Canada  
1  Québec  Canada  
2  Québec  Canada  
3  Québec  Canada  
4  Québec  Canada

On peut passer en argument le nombre de lignes qu'on veut lire. Par exemple pour lire les 10 premières lignes on peut taper la commande suivante:

In [10]:
df1.head(10)

num  annee    prenom           nom genre          specialite1  \
0  #30001   1930    Lucien         Amyot     M  Médecine de famille   
1  #30002   1930   Wilfrid         Audet     M             Inconnue   
2  #30003   1930     A. T.       Batshaw     M  Médecine de famille   
3  #30005   1930   Maurice       Beaudry     M         Dermatologie   
4  #30006   1930     P. E.       Belleau     M     Médecine interne   
5  #30008   1930    Antoni         Blais     M  Médecine de famille   
6  #30012   1930      Lyla         Brown     F  Médecine de famille   
7  #30014   1930  Alphonse     Couturier     M   Chirurgie générale   
8  #30015   1930       Guy  D'Argencourt     M   Chirurgie générale   
9  #30017   1930      Rémi    Desjardins     M  Médecine de famille   

  specialite2 specialite3                                 statut  \
0         NaN         NaN                     Retrait pour décès   
1         NaN         NaN                              Démission   
2         NaN         NaN                     Retrait pour décès   
3         NaN         NaN                     Retrait pour décès   
4         NaN         NaN                     Retrait pour décès   
5         NaN         NaN                     Retrait pour décès   
6         NaN         NaN                     Retrait pour décès   
7         NaN         NaN                     Retrait pour décès   
8         NaN         NaN  Radié pour non-paiement de cotisation   
9         NaN         NaN                     Retrait pour décès   

  dateChangStatut typePermis           ville    prov    pays  
0      1981-12-29   Régulier        Montréal  Québec  Canada  
1      1987-10-01   Régulier            Alma  Québec  Canada  
2      1983-11-11   Régulier        Montréal  Québec  Canada  
3      1989-02-12   Régulier          Québec  Québec  Canada  
4      1986-05-31   Régulier          Québec  Québec  Canada  
5      1983-06-30   Régulier      Sherbrooke  Québec  Canada  
6      1994-06-07   Régulier        Montréal  Québec  Canada  
7      1995-08-11   Régulier       Cap-Rouge  Québec  Canada  
8      1988-11-30   Régulier       Outremont  Québec  Canada  
9      1989-09-23   Régulier  Sainte-Thérèse  Québec  Canada

## De la même façon pour lire les dernières lignes du dataframe on peut utiliser la fonction tail(), qui par défaut afficher les cinq dernières lignes de notre dataframeframe

In [12]:
df1.tail() # on peut aussi spécifier le nombre de lignes qu'on peut lire 

num  annee                  prenom       nom genre  \
35104  #16893   2016               Sébastien     David     M   
35105  #16894   2016  Guillaume Patrick Jean  Berthier     M   
35106  #16895   2016                  Miriam    Banoub     F   
35107  #16896   2016             Anne-Sophie     Lemay     F   
35108  #16897   2016             Mohamed Ali      Abla     M   

               specialite1         specialite2 specialite3           statut  \
35104  Médecine de famille                 NaN         NaN  Inscrit - Actif   
35105  Médecine de famille                 NaN         NaN  Inscrit - Actif   
35106  Médecine de famille                 NaN         NaN  Inscrit - Actif   
35107          Hématologie  Oncologie médicale         NaN  Inscrit - Actif   
35108  Médecine de famille                 NaN         NaN  Inscrit - Actif   

      dateChangStatut  typePermis ville    prov    pays  
35104             NaN    Régulier   NaN  Québec  Canada  
35105             NaN  Restrictif   NaN  Québec  Canada  
35106             NaN    Régulier   NaN  Québec  Canada  
35107             NaN    Régulier   NaN  Québec  Canada  
35108             NaN    Régulier   NaN  Québec  Canada

In [8]:
df2.head()  # deuxieme dataframe

annee  trimestre        date                    fournisseur  \
0   2004          1  01-04-2004           Tri-Comp Systems Ltd   
1   2004          1  01-04-2004  Associates Group of Companies   
2   2004          1  01-04-2004           InBusiness Solutions   
3   2004          1  01-04-2004   Maple Leaf Construction Ltd.   
4   2004          1  01-04-2004  Wade General Contracting Ltd.   

                                         description   montant  
0              Services professionnels, informatique  186328.0  
1                          Services de programmation   84263.0  
2                          Services de programmation   88730.0  
3  Travaux de construction d'ouvrages de génie civil  448375.0  
4  Travaux d'entreprises de construction spéciali...  217816.0

In [13]:
df2.tail()

annee  trimestre        date                         fournisseur  \
165947   2016          3  30-12-2016                     NB Harrison Inc   
165948   2016          3  30-12-2016                       Clint Clawson   
165949   2016          3  30-12-2016  The Goodyear Tire & Rubber Company   
165950   2016          3  30-12-2016                Imperial Oil Limited   
165951   2016          3  30-12-2016        S.M.I. Support Measures Inc.   

                                              description   montant  
165947                          Opérations de parachutage   30700.0  
165948                          Opérations de parachutage   41250.0  
165949          Pneumatiques et chambres à air d'aéronefs  282000.0  
165950                             Combustible d'aviation  315315.2  
165951  Antennes, guides d'ondes et équipement connexe...  630000.0

pour connaitre le nombre de lignes et le nombre de colonnes on utilise la fonction shape comme suit:

In [14]:
df1.shape

(35109, 14)

Comme nous le voyons, notre premier dataframe comporte 35109 lignes et 14 colonnes

## Si on veut connaitre uniquement le nombre de lignes on peut taper la commande suivante:

In [16]:
df1.shape[0] 

35109

## Si on veut connaitre uniquement le nombre de colonnes on peut taper la commande suivante:

In [17]:
df1.shape[1]

14

In [18]:
df2.shape

(165952, 6)

Pour lister les noms des colonnes on tape la commande suivante:

In [19]:
df1.columns

Index(['num', 'annee', 'prenom', 'nom', 'genre', 'specialite1', 'specialite2',
       'specialite3', 'statut', 'dateChangStatut', 'typePermis', 'ville',
       'prov', 'pays'],
      dtype='object')

On peut aussi utiliser la fonction len pour connaitre le nombre de colonnes comme suit

In [20]:
len(df1.columns) 

14

In [22]:
df2.columns

Index(['annee', 'trimestre', 'date', 'fournisseur', 'description', 'montant'], dtype='object')

## Pour avoir beaucoup plus d'informations sur notre dataframe, on peut utiliser la fonction info(). Cette fonction nous permet de connaitre le nombre de lignes, de colonnes, les valeurs manquantes, le memoire utilisé, etc.

In [23]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35109 entries, 0 to 35108
Data columns (total 14 columns):
num                35109 non-null object
annee              35109 non-null int64
prenom             35108 non-null object
nom                35109 non-null object
genre              35109 non-null object
specialite1        35109 non-null object
specialite2        3467 non-null object
specialite3        224 non-null object
statut             35109 non-null object
dateChangStatut    11412 non-null object
typePermis         35109 non-null object
ville              33715 non-null object
prov               34133 non-null object
pays               35109 non-null object
dtypes: int64(1), object(13)
memory usage: 3.8+ MB


In [25]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 165952 entries, 0 to 165951
Data columns (total 6 columns):
annee          165952 non-null int64
trimestre      165952 non-null int64
date           165952 non-null object
fournisseur    165952 non-null object
description    165945 non-null object
montant        165952 non-null float64
dtypes: float64(1), int64(2), object(3)
memory usage: 7.6+ MB


## Pour compter le nombre d'éléments de chaque colonne, on utilise la fonction count comme suit:

In [26]:
df1.count()

num                35109
annee              35109
prenom             35108
nom                35109
genre              35109
specialite1        35109
specialite2         3467
specialite3          224
statut             35109
dateChangStatut    11412
typePermis         35109
ville              33715
prov               34133
pays               35109
dtype: int64

Ici on remarque que la colonne specialite3 contient seulement 224 lignes non vides.

In [27]:
df2.count()

annee          165952
trimestre      165952
date           165952
fournisseur    165952
description    165945
montant        165952
dtype: int64

## Pour compter le nombre de valeurs manquantes de chaque colonne on peut utiliser la fonction isnull suivie de sum

In [28]:
df1.isnull().sum()

num                    0
annee                  0
prenom                 1
nom                    0
genre                  0
specialite1            0
specialite2        31642
specialite3        34885
statut                 0
dateChangStatut    23697
typePermis             0
ville               1394
prov                 976
pays                   0
dtype: int64

La colonne specialit3 contient 34885 valeurs manquantes.

In [32]:
df2.isna().sum() # On peut aussi utiliser la fonction isna() suivie de sum().

annee          0
trimestre      0
date           0
fournisseur    0
description    7
montant        0
dtype: int64

## Il est important de connaitre le type de chaque colonne de notre dataframe. Pour ce faire, on utilise la fonction dtypes:

In [33]:
df1.dtypes

num                object
annee               int64
prenom             object
nom                object
genre              object
specialite1        object
specialite2        object
specialite3        object
statut             object
dateChangStatut    object
typePermis         object
ville              object
prov               object
pays               object
dtype: object

In [34]:
df2.dtypes

annee            int64
trimestre        int64
date            object
fournisseur     object
description     object
montant        float64
dtype: object

## Pour afficher les 30 premières lignes et les 30 dernières colones, on peut utilisier la fonction print

In [37]:
print(df1)

          num  annee                       prenom                nom genre  \
0      #30001   1930                       Lucien              Amyot     M   
1      #30002   1930                      Wilfrid              Audet     M   
2      #30003   1930                        A. T.            Batshaw     M   
3      #30005   1930                      Maurice            Beaudry     M   
4      #30006   1930                        P. E.            Belleau     M   
5      #30008   1930                       Antoni              Blais     M   
6      #30012   1930                         Lyla              Brown     F   
7      #30014   1930                     Alphonse          Couturier     M   
8      #30015   1930                          Guy       D'Argencourt     M   
9      #30017   1930                         Rémi         Desjardins     M   
10     #30020   1930                       Armand           Frappier     M   
11     #30021   1930                       Edmour           Fren

In [39]:
df1[:] # permet d'obtenir à peu près la même chose que précédemment

num  annee                       prenom                nom genre  \
0      #30001   1930                       Lucien              Amyot     M   
1      #30002   1930                      Wilfrid              Audet     M   
2      #30003   1930                        A. T.            Batshaw     M   
3      #30005   1930                      Maurice            Beaudry     M   
4      #30006   1930                        P. E.            Belleau     M   
5      #30008   1930                       Antoni              Blais     M   
6      #30012   1930                         Lyla              Brown     F   
7      #30014   1930                     Alphonse          Couturier     M   
8      #30015   1930                          Guy       D'Argencourt     M   
9      #30017   1930                         Rémi         Desjardins     M   
10     #30020   1930                       Armand           Frappier     M   
11     #30021   1930                       Edmour           Frenette     M   
12     #30024   1930                        I. M.           Gauthier     M   
13     #30028   1930                        J. A.             Hébert     M   
14     #30029   1930                       Pierre             Jodoin     M   
15     #30030   1930                        J. A.             Kearns     M   
16     #30032   1930                      J.-Paul           Laplante     M   
17     #30034   1930                           R.         Latourelle     M   
18     #30039   1930                         Jean             Matton     M   
19     #30042   1930                       Marcel            Ostiguy     M   
20     #30045   1930                       Marthe            Pelland     F   
21     #30047   1930                    Lucien F.           Peloquin     M   
22     #30048   1930                       Lucien         Petitclerc     M   
23     #30050   1930                        L. P.           Poliquin     M   
24     #30052   1930                      Antonio             Robert     M   
25     #30056   1930                       Claude            St-Onge     M   
26     #30057   1930                         Paul            St-Onge     M   
27     #30058   1930                   Antonotien          St-Pierre     M   
28     #30060   1930                        Allen             Temple     M   
29     #30062   1930                        Edgar            Vanasse     M   
...       ...    ...                          ...                ...   ...   
35079  #16863   2016                        Siham          El Housny     F   
35080  #16864   2016                    Geneviève              Hardy     F   
35081  #16865   2016                      Cynthia              Lemay     F   
35082  #16866   2016                Jean-François          Malenfant     M   
35083  #16867   2016                         Mona        Abderrahman     F   
35084  #16868   2016             Emmanuel Patrice             Valcin     M   
35085  #16869   2016                  Cesar Pavel       Osores Tello     M   
35086  #16870   2016                    Alexandra             Fortin     F   
35087  #16871   2016                     Philippe           Melanson     M   
35088  #16872   2016                       Judith        Brouillette     F   
35089  #16874   2016                        Aimée           Duquette     F   
35090  #16875   2016                        David           Puissant     M   
35091  #16876   2016                    Catherine         Robitaille     F   
35092  #16877   2016                      Meriame               Dami     F   
35093  #16878   2016                        Ardit            Haxhija     M   
35094  #16879   2016                         Rami           Bensalma     M   
35095  #16880   2016                  Zulma Erika  Sotomayor Alvarez     F   
35096  #16881   2016                       Amanda            Violato     F   
35097  #16882   2016                        Josée              Larin     F   
35098  #16883   2016            

## il est possible de supprimer une colonne en utilisant la fonction drop en lui donnant comment paramétre la fonction qu'on peut supprimer et de mettre axis = 1

In [67]:
df3 = df1.drop('specialite3', axis = 1)

In [68]:
df3.columns

Index(['num', 'annee', 'prenom', 'nom', 'genre', 'specialite1', 'specialite2',
       'statut', 'dateChangStatut', 'typePermis', 'ville', 'prov', 'pays'],
      dtype='object')

In [43]:
df3.shape

(35109, 13)

On voit que le dataframe df3 contient seulement 13 colonnes contrairement au dataframe df1 qui en contient 14

# Filtre

### Il est aussi possible de faire des filtres. Par exemole on peut créer un autre dataframe appelé df4 à partir de notre dataframe df1 qui contient unique les medcins de sexe masculine

In [44]:
df4 = df1[df1.genre == 'M']

In [45]:
df4.count()

num                22441
annee              22441
prenom             22440
nom                22441
genre              22441
specialite1        22441
specialite2         2318
specialite3          143
statut             22441
dateChangStatut     9533
typePermis         22441
ville              21602
prov               21613
pays               22441
dtype: int64

### De même pour df5 qui contient uniquement les médecins de sexes féminin

In [46]:
df5 = df1[df1.genre == 'F'] 

In [47]:
df5. info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12668 entries, 6 to 35107
Data columns (total 14 columns):
num                12668 non-null object
annee              12668 non-null int64
prenom             12668 non-null object
nom                12668 non-null object
genre              12668 non-null object
specialite1        12668 non-null object
specialite2        1149 non-null object
specialite3        81 non-null object
statut             12668 non-null object
dateChangStatut    1879 non-null object
typePermis         12668 non-null object
ville              12113 non-null object
prov               12520 non-null object
pays               12668 non-null object
dtypes: int64(1), object(13)
memory usage: 1.4+ MB


In [48]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 165952 entries, 0 to 165951
Data columns (total 6 columns):
annee          165952 non-null int64
trimestre      165952 non-null int64
date           165952 non-null object
fournisseur    165952 non-null object
description    165945 non-null object
montant        165952 non-null float64
dtypes: float64(1), int64(2), object(3)
memory usage: 7.6+ MB


# Calcul statistique

#### On peut utiliser les fonction min, max, median, etc pour déterminer la valeur minimale, maximale, etc 

In [49]:
df2['montant'].min() # valeur minimale de la colonne montant

0.0

In [50]:
df2['montant'].max() # valeur maximale de la colonne montant

9975000.0

In [52]:
df2['montant'].mean() # valeur moyenne de la colonne montant

163681.63551063047

### On peut obtenir toutes ces informations en utilisant la méthode describe 

In [56]:
df2['montant'].describe()

count    1.659520e+05
mean     1.636816e+05
std      5.958607e+05
min      0.000000e+00
25%      1.575475e+04
50%      2.898450e+04
75%      8.623100e+04
max      9.975000e+06
Name: montant, dtype: float64

#### la méthode describe par défaut agit les colonnes de types numériques. pour l'uttiliser sur toutes les colonnes on la passe argument include = "all"

In [58]:
df1.describe()

annee
count  35109.000000
mean    1984.805691
std       20.449279
min     1930.000000
25%     1971.000000
50%     1984.000000
75%     2003.000000
max     2016.000000

In [59]:
df1.describe(include="all")

num         annee  prenom    nom  genre          specialite1  \
count    35109  35109.000000   35108  35109  35109                35109   
unique   35109           NaN   10457  13010      2                   82   
top     #85557           NaN  Pierre    Roy      M  Médecine de famille   
freq         1           NaN     621    239  22441                13150   
mean       NaN   1984.805691     NaN    NaN    NaN                  NaN   
std        NaN     20.449279     NaN    NaN    NaN                  NaN   
min        NaN   1930.000000     NaN    NaN    NaN                  NaN   
25%        NaN   1971.000000     NaN    NaN    NaN                  NaN   
50%        NaN   1984.000000     NaN    NaN    NaN                  NaN   
75%        NaN   2003.000000     NaN    NaN    NaN                  NaN   
max        NaN   2016.000000     NaN    NaN    NaN                  NaN   

        specialite2         specialite3           statut dateChangStatut  \
count          3467                 224            35109           11412   
unique           65                  30               14            5114   
top     Cardiologie  Oncologie médicale  Inscrit - Actif      2015-09-10   
freq            385                  82            21017             120   
mean            NaN                 NaN              NaN             NaN   
std             NaN                 NaN              NaN             NaN   
min             NaN                 NaN              NaN             NaN   
25%             NaN                 NaN              NaN             NaN   
50%             NaN                 NaN              NaN             NaN   
75%             NaN                 NaN              NaN             NaN   
max             NaN                 NaN              NaN             NaN   

       typePermis     ville    prov    pays  
count       35109     33715   34133   35109  
unique          3      2206      61      60  
top      Régulier  Montréal  Québec  Canada  
freq        33781      8054   29794   32737  
mean          NaN       NaN     NaN     NaN  
std           NaN       NaN     NaN     NaN  
min           NaN       NaN     NaN     NaN  
25%           NaN       NaN     NaN     NaN  
50%           NaN       NaN     NaN     NaN  
75%           NaN       NaN     NaN     NaN  
max           NaN       NaN     NaN     NaN

## On peut déterminer aussi le montant total en utilisant la méthode sum

In [57]:
df2['montant'].sum() 

27163294776.26

# Modification d'un  dataframe

In [60]:
df3

num  annee                       prenom                nom genre  \
0      #30001   1930                       Lucien              Amyot     M   
1      #30002   1930                      Wilfrid              Audet     M   
2      #30003   1930                        A. T.            Batshaw     M   
3      #30005   1930                      Maurice            Beaudry     M   
4      #30006   1930                        P. E.            Belleau     M   
5      #30008   1930                       Antoni              Blais     M   
6      #30012   1930                         Lyla              Brown     F   
7      #30014   1930                     Alphonse          Couturier     M   
8      #30015   1930                          Guy       D'Argencourt     M   
9      #30017   1930                         Rémi         Desjardins     M   
10     #30020   1930                       Armand           Frappier     M   
11     #30021   1930                       Edmour           Frenette     M   
12     #30024   1930                        I. M.           Gauthier     M   
13     #30028   1930                        J. A.             Hébert     M   
14     #30029   1930                       Pierre             Jodoin     M   
15     #30030   1930                        J. A.             Kearns     M   
16     #30032   1930                      J.-Paul           Laplante     M   
17     #30034   1930                           R.         Latourelle     M   
18     #30039   1930                         Jean             Matton     M   
19     #30042   1930                       Marcel            Ostiguy     M   
20     #30045   1930                       Marthe            Pelland     F   
21     #30047   1930                    Lucien F.           Peloquin     M   
22     #30048   1930                       Lucien         Petitclerc     M   
23     #30050   1930                        L. P.           Poliquin     M   
24     #30052   1930                      Antonio             Robert     M   
25     #30056   1930                       Claude            St-Onge     M   
26     #30057   1930                         Paul            St-Onge     M   
27     #30058   1930                   Antonotien          St-Pierre     M   
28     #30060   1930                        Allen             Temple     M   
29     #30062   1930                        Edgar            Vanasse     M   
...       ...    ...                          ...                ...   ...   
35079  #16863   2016                        Siham          El Housny     F   
35080  #16864   2016                    Geneviève              Hardy     F   
35081  #16865   2016                      Cynthia              Lemay     F   
35082  #16866   2016                Jean-François          Malenfant     M   
35083  #16867   2016                         Mona        Abderrahman     F   
35084  #16868   2016             Emmanuel Patrice             Valcin     M   
35085  #16869   2016                  Cesar Pavel       Osores Tello     M   
35086  #16870   2016                    Alexandra             Fortin     F   
35087  #16871   2016                     Philippe           Melanson     M   
35088  #16872   2016                       Judith        Brouillette     F   
35089  #16874   2016                        Aimée           Duquette     F   
35090  #16875   2016                        David           Puissant     M   
35091  #16876   2016                    Catherine         Robitaille     F   
35092  #16877   2016                      Meriame               Dami     F   
35093  #16878   2016                        Ardit            Haxhija     M   
35094  #16879   2016                         Rami           Bensalma     M   
35095  #16880   2016                  Zulma Erika  Sotomayor Alvarez     F   
35096  #16881   2016                       Amanda            Violato     F   
35097  #16882   2016                        Josée              Larin     F   
35098  #16883   2016            

In [69]:
df3.columns

Index(['num', 'annee', 'prenom', 'nom', 'genre', 'specialite1', 'specialite2',
       'statut', 'dateChangStatut', 'typePermis', 'ville', 'prov', 'pays'],
      dtype='object')

### Notre dataframe df3 contient 13 colonnes: vérification avec la fonction len

In [70]:
len(df3.columns)

13

## Ici on supprime la colonne specialite1

In [62]:
df3.drop('specialite1', axis = 1)

num  annee                       prenom                nom genre  \
0      #30001   1930                       Lucien              Amyot     M   
1      #30002   1930                      Wilfrid              Audet     M   
2      #30003   1930                        A. T.            Batshaw     M   
3      #30005   1930                      Maurice            Beaudry     M   
4      #30006   1930                        P. E.            Belleau     M   
5      #30008   1930                       Antoni              Blais     M   
6      #30012   1930                         Lyla              Brown     F   
7      #30014   1930                     Alphonse          Couturier     M   
8      #30015   1930                          Guy       D'Argencourt     M   
9      #30017   1930                         Rémi         Desjardins     M   
10     #30020   1930                       Armand           Frappier     M   
11     #30021   1930                       Edmour           Frenette     M   
12     #30024   1930                        I. M.           Gauthier     M   
13     #30028   1930                        J. A.             Hébert     M   
14     #30029   1930                       Pierre             Jodoin     M   
15     #30030   1930                        J. A.             Kearns     M   
16     #30032   1930                      J.-Paul           Laplante     M   
17     #30034   1930                           R.         Latourelle     M   
18     #30039   1930                         Jean             Matton     M   
19     #30042   1930                       Marcel            Ostiguy     M   
20     #30045   1930                       Marthe            Pelland     F   
21     #30047   1930                    Lucien F.           Peloquin     M   
22     #30048   1930                       Lucien         Petitclerc     M   
23     #30050   1930                        L. P.           Poliquin     M   
24     #30052   1930                      Antonio             Robert     M   
25     #30056   1930                       Claude            St-Onge     M   
26     #30057   1930                         Paul            St-Onge     M   
27     #30058   1930                   Antonotien          St-Pierre     M   
28     #30060   1930                        Allen             Temple     M   
29     #30062   1930                        Edgar            Vanasse     M   
...       ...    ...                          ...                ...   ...   
35079  #16863   2016                        Siham          El Housny     F   
35080  #16864   2016                    Geneviève              Hardy     F   
35081  #16865   2016                      Cynthia              Lemay     F   
35082  #16866   2016                Jean-François          Malenfant     M   
35083  #16867   2016                         Mona        Abderrahman     F   
35084  #16868   2016             Emmanuel Patrice             Valcin     M   
35085  #16869   2016                  Cesar Pavel       Osores Tello     M   
35086  #16870   2016                    Alexandra             Fortin     F   
35087  #16871   2016                     Philippe           Melanson     M   
35088  #16872   2016                       Judith        Brouillette     F   
35089  #16874   2016                        Aimée           Duquette     F   
35090  #16875   2016                        David           Puissant     M   
35091  #16876   2016                    Catherine         Robitaille     F   
35092  #16877   2016                      Meriame               Dami     F   
35093  #16878   2016                        Ardit            Haxhija     M   
35094  #16879   2016                         Rami           Bensalma     M   
35095  #16880   2016                  Zulma Erika  Sotomayor Alvarez     F   
35096  #16881   2016                       Amanda            Violato     F   
35097  #16882   2016                        Josée              Larin     F   
35098  #16883   2016            

### Après suppression de la colonne spécialité, le nombre de colonnes de df3 ne change pas. Pour apporter ce changement, on passe en argument la fonction drop inplace = True

In [63]:
len(df3.columns)

13

In [71]:
df3.drop('specialite1', axis = 1, inplace=True)

In [72]:
len(df3.columns)

12